extensions/wikia/UserLogin/UserLoginSpecialController.class.php

```diff
@@ -114,286 +114,86 @@ private function getAuthenticationResource() {
-				$this->msg = $response->getVal( 'msg', '' );
-				$this->errParam = $response->getVal( 'errParam', '' );
-
-				// set the language object
-				$code = $this->wg->request->getVal( 'uselang', $this->wg->User->getGlobalPreference( 'language' ) );
-				$this->wg->Lang = Language::factory( $code );
-
-				if ( $this->result == 'ok' ) {
-					// redirect page
-					$this->userLoginHelper->doRedirect();
-				} elseif ( $this->result == 'unconfirm' ) {
-					$response = $this->app->sendRequest(
-						'UserLoginSpecial',
-						'getUnconfirmedUserRedirectUrl',
-						[
-							'username' => $this->username,
-							'uselang' => $code
-						]
-					);
-					$redirectUrl = $response->getVal( 'redirectUrl' );
-					$this->wg->out->redirect( $redirectUrl );
-				} elseif ( $this->result === 'closurerequested' ) {
-					$response = $this->app->sendRequest( 'UserLoginSpecial', 'getCloseAccountRedirectUrl' );
-					$redirectUrl = $response->getVal( 'redirectUrl' );
-					$this->wg->Out->redirect( $redirectUrl );
-				} elseif ( $this->result == 'resetpass' ) {
-					$this->editToken = $this->wg->User->getEditToken();
-					$this->subheading = wfMessage( 'resetpass_announce' )->escaped();
-					$this->wg->Out->setPageTitle( wfMessage( 'userlogin-password-page-title' )->plain() );
-					$this->response->getView()->setTemplate( 'UserLoginSpecial', 'changePassword' );
-				}
+				return true;
 			}
-		}
 
-		if ( $type === 'forgotPassword' ) {
-			$this->overrideTemplate( 'forgotPassword' );
-			// set page title
-			$this->wg->Out->setPageTitle( wfMessage( 'userlogin-forgot-password' )->plain() );
-			return true;
-		}
-
-		// we're sure at this point we'll need the private fields'
-		// values in the template let's pass them then
-		$this->response->setVal( 'username', $this->username );
-		$this->response->setVal( 'password', $this->password );
-
-		if ( $this->app->checkSkin( 'wikiamobile' ) ) {
-			$recoverParam = 'recover=1';
-			$recover = ( $this->wg->request->getInt( 'recover' ) === 1 );
+			if ( $this->result === 'closurerequested' ) {
+				$response =
+					$this->app->sendRequest( 'UserLoginSpecial', 'getCloseAccountRedirectUrl' );
+				$redirectUrl = $response->getVal( 'redirectUrl' );
+				$this->wg->Out->redirect( $redirectUrl );
 
-			$this->response->setVal( 'recoverLink', SpecialPage::getTitleFor( 'UserLogin' )->getLocalURL( $recoverParam ) );
-			$this->response->setVal( 'recoverPassword', $recover );
-
-			if ( $recover ) {
-				// we use different AssetsManager packages for recover and login, so make sure the page URL is different to avoid Varnish clashes
-				$this->formPostAction .= "?{$recoverParam}";
-			}
-
-			if ( !empty( $action ) && ( $action === wfMessage( 'resetpass_submit' )->escaped()  || $this->result == 'resetpass' ) ) {
-				$this->overrideTemplate( 'WikiaMobileChangePassword' );
-			} else {
-				$this->overrideTemplate( 'WikiaMobileIndex' );
+				return true;
 			}
 		}
 
-		return true;
+		$this->redirectToMercury();
```

Comment:
And finally redirect to mercury if not mobile change password nor change password nor account closure.